.github/workflows/docker_build.yml

```diff
@@ -0,0 +1,64 @@
+name: Build docker image and push to GHCR
+
+on:
+  push:
```

Comment:
we can change this to be daily or whatever you'd like

Reply:
Can you quickly explain if the published docker image contains a preinstalled version of mathlib or not? (I don't quite understand and that would decide whether it should be run on a cron or not)..

It looks to me like the current setup has no specific mathlib version installed (reading `gitpod/DOCKERFILE`). @YaelDillies's personally maintained container mentioned in #23530 doesn't either. Do I understand correctly that this CI workflow does create a docker container with a specific mathlib version?

If so, why wouldn't you already download cache and everything inside the pre-built docker container? (as far as I understand that only happens on opening gitpod, doesn't it?

Reply:
There are three layers to the Docker setup:
* The Docker**file** contains the instructions for building...
* ... the Docker **image** which is an OS image that can be deployed inside...
* ... a Docker **container**, which is free to run extra commands on top

Building the Docker image from the Dockerfile is currently done manually by me. The current PR proposes to do it automatically.

Checking out a copy of the mathlib4 repo is the duty of the Docker container. You can check out other repos too with the same image. Put simply, your question amounts to asking why Microsoft is selling Windows without a version of mathlib installed.

Reply:
> we can change this to be daily or whatever you'd like

Coming back to this, yes, I think it's worth building the Docker image only once a day. It changes at most once a month anyway, so building it on every push feels excessive